packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -111,12 +157,104 @@ export class PathNode extends PathShared<FieldKey> {
  * Thus this can be thought of as a sparse copy of the subset of trees which are used as anchors
  * (and thus need parent paths).
  */
-export class PathCollection extends PathShared<RootField> {
-    public constructor() {
-        super();
+class PathNode implements UpPath {
+    /**
+     * Number of references to this from external sources (ex: `Anchors` via `AnchorSet`.)
+     */
+    private refCount = 1;
+
+    /**
+     * Construct a PathNode with refcount 1.
+     * @param anchorSet - used to determine if this PathNode is already part a specific anchorSet
+     * to early out UpPath walking.
+     */
+    public constructor(
+        public readonly anchorSet: AnchorSet,
+        public parentField: FieldKey,
+        public parentIndex: number,
+        public parentPath: PathNode | undefined) {}
+    // PathNode arrays are kept sorted by index for efficient search.
+    protected readonly children: Map<FieldKey, PathNode[]> = new Map();
+    // public constructor() {}
+
+    public get parent(): UpPath | undefined {
+        // Root PathNode corresponds to the undefined root for UpPath.
+        if (this.parentPath?.parentPath === undefined) {
+            return undefined;
+        }
+        return this.parentPath;
+    }
+
+    public addRef(count = 1): void {
+        this.refCount += count;
+    }
+
+    public removeRef(count = 1): void {
+        this.refCount -= count;
+        if (this.refCount < 1) {
+            assert(this.refCount === 0, "PathNode Refcount should not be negative.");
+
+            if (this.children.size === 0) {
+                this.deleteThis();
+            }
+        }
+    }
+
+    /**
+     * Gets a child, adding a ref to it.
+     * Creates child (with 1 ref) if needed.
+     */
+    public getOrCreateChild(key: FieldKey, index: number): PathNode {
+        let field = this.children.get(key);
+        if (field === undefined) {
+            field = [];
+            this.children.set(key, field);
+        }
+        // TODO: should do more optimized search (ex: binary search or better) using index;
+        // Note that this is the index in the list of child paths, not the index withing the field
```

Comment:
This comment looks like it may have been copied from `removeChild`, but I don't think it's true here.